### **PREDICT NET INCOME FOR THE YEAR 2020**

In [42]:
import warnings
warnings.filterwarnings('ignore')

import pandas as pd

data = pd.read_csv('canada_per_capita_income.csv')
data

year  per capita income (US$)
0   1970              3399.299037
1   1971              3768.297935
2   1972              4251.175484
3   1973              4804.463248
4   1974              5576.514583
5   1975              5998.144346
6   1976              7062.131392
7   1977              7100.126170
8   1978              7247.967035
9   1979              7602.912681
10  1980              8355.968120
11  1981              9434.390652
12  1982              9619.438377
13  1983             10416.536590
14  1984             10790.328720
15  1985             11018.955850
16  1986             11482.891530
17  1987             12974.806620
18  1988             15080.283450
19  1989             16426.725480
20  1990             16838.673200
21  1991             17266.097690
22  1992             16412.083090
23  1993             15875.586730
24  1994             15755.820270
25  1995             16369.317250
26  1996             16699.826680
27  1997             17310.757750
28  1998             16622.671870
29  1999             17581.024140
30  2000             18987.382410
31  2001             18601.397240
32  2002             19232.175560
33  2003             22739.426280
34  2004             25719.147150
35  2005             29198.055690
36  2006             32738.262900
37  2007             36144.481220
38  2008             37446.486090
39  2009             32755.176820
40  2010             38420.522890
41  2011             42334.711210
42  2012             42665.255970
43  2013             42676.468370
44  2014             41039.893600
45  2015             35175.188980
46  2016             34229.193630

This dataset shows Canada's capital income for the years, 1970 to 2016. <br> There are no missing values and no duplicates as can be observed. <br> I want to predict for the years, 2017 to 2025.

In [43]:
data.shape

(47, 2)

In [44]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 47 entries, 0 to 46
Data columns (total 2 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   year                     47 non-null     int64  
 1   per capita income (US$)  47 non-null     float64
dtypes: float64(1), int64(1)
memory usage: 880.0 bytes


In [45]:
data.duplicated().sum()

0

In [46]:
data.isna().sum()

year                       0
per capita income (US$)    0
dtype: int64

In [47]:
data.describe()

year  per capita income (US$)
count    47.000000                47.000000
mean   1993.000000             18920.137063
std      13.711309             12034.679438
min    1970.000000              3399.299037
25%    1981.500000              9526.914515
50%    1993.000000             16426.725480
75%    2004.500000             27458.601420
max    2016.000000             42676.468370

In [48]:
y = data['per capita income (US$)']
X = data.drop('per capita income (US$)', axis=1)

In [49]:
from sklearn.linear_model import LinearRegression

linreg = LinearRegression()
linreg.fit(data[['year']], data['per capita income (US$)'])

LinearRegression()

In [50]:
linreg.predict([[2020]])

array([41288.69409442])

In [51]:
#y=Mx+B y=Income(Dependent variable) M=Gradient/Slope x=year(Independent Variable) B=Intercept

linreg.coef_ #M

array([828.46507522])

In [52]:
linreg.intercept_ #B

-1632210.7578554575

In [53]:
years = pd.read_csv('income_prediction.csv')
years

year
0  2017
1  2018
2  2019
3  2020
4  2021
5  2022
6  2023
7  2024
8  2025

Here, i load the csv file with the years i want to predict the incomes.

In [54]:
prediction = linreg.predict(years)
prediction

array([38803.29886875, 39631.76394397, 40460.22901919, 41288.69409442,
       42117.15916964, 42945.62424486, 43774.08932009, 44602.55439531,
       45431.01947053])

In [55]:
years['income (US$)'] = prediction
years

year  income (US$)
0  2017  38803.298869
1  2018  39631.763944
2  2019  40460.229019
3  2020  41288.694094
4  2021  42117.159170
5  2022  42945.624245
6  2023  43774.089320
7  2024  44602.554395
8  2025  45431.019471

I then create a new column called 'income' where the predicted values are going to be filled based on the years.

In [56]:
years.to_csv('2017-2025 Incomes.csv', index=False)

Then saved the prediction to a csv file.

## **CONCLUSION**
For the year 2020, the capital income is $41288.694094.